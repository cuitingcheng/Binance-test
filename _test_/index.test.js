/* eslint-env jest */

import { shallow, mount } from 'enzyme'
import React from 'react'
import renderer from 'react-test-renderer'

import Index from '../pages/index.js'
import Trade from '../pages/trade.js'
const data = [{"symbol":"ETHSGD","quoteAssetName":"SGD","tradedMoney":357.0,"baseAssetUnit":"","baseAssetName":"ETH","baseAsset":"ETH","tickSize":"0.0000001","prevClose":140.0,"activeBuy":0.0,"high":"140.0000000","lastAggTradeId":-1,"low":"140.0000000","matchingUnitType":"STANDARD","close":"140.0000000","quoteAsset":"SGD","productType":null,"active":true,"minTrade":0.01000000,"activeSell":2.55,"withdrawFee":"10","volume":"2.5500000","decimalPlaces":8,"quoteAssetUnit":"","open":"140.0000000","status":"TRADING","minQty":1E-8},{"symbol":"BTCSGD","quoteAssetName":"SGD","tradedMoney":180.6,"baseAssetUnit":"฿","baseAssetName":"Bitcoin","baseAsset":"BTC","tickSize":"0.01","prevClose":53.7,"activeBuy":0.0,"high":"53.70","lastAggTradeId":-1,"low":"1.10","matchingUnitType":"STANDARD","close":"5.00","quoteAsset":"SGD","productType":null,"active":true,"minTrade":0.00000100,"activeSell":57.0,"withdrawFee":"10","volume":"57.00","decimalPlaces":8,"quoteAssetUnit":"","open":"53.70","status":"TRADING","minQty":1E-8}];
const cols = Object.keys(data[0]);

describe('With Enzyme', () => {
  it('h1 shows "可交易货币"', () => {
    const app = shallow(<Index data={data} cols={cols} />);

    expect(app.find('h1').text()).toEqual('可交易货币');
  })
});

describe('With Snapshot Testing', () => {
  it('Index shows list with example data', () => {
    const component = renderer.create(<Index data={data} cols={cols} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })
});

describe('Route without Symbol', () => {
  it('Trade shows "No Symbol."', () => {
    const app = mount(<Trade router={{query:{symbol:null}}}/>);
    expect(app.find('p').text()).toEqual('No Symbol.');
  })
});
