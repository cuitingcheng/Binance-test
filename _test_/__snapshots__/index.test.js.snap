// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing Index shows list with example data 1`] = `
<div>
  <h1>
    可交易货币
  </h1>
  <table>
    <tbody>
      <tr>
        <th>
          symbol
        </th>
        <th>
          quoteAssetName
        </th>
        <th>
          tradedMoney
        </th>
        <th>
          baseAssetUnit
        </th>
        <th>
          baseAssetName
        </th>
        <th>
          baseAsset
        </th>
        <th>
          tickSize
        </th>
        <th>
          prevClose
        </th>
        <th>
          activeBuy
        </th>
        <th>
          high
        </th>
        <th>
          lastAggTradeId
        </th>
        <th>
          low
        </th>
        <th>
          matchingUnitType
        </th>
        <th>
          close
        </th>
        <th>
          quoteAsset
        </th>
        <th>
          productType
        </th>
        <th>
          active
        </th>
        <th>
          minTrade
        </th>
        <th>
          activeSell
        </th>
        <th>
          withdrawFee
        </th>
        <th>
          volume
        </th>
        <th>
          decimalPlaces
        </th>
        <th>
          quoteAssetUnit
        </th>
        <th>
          open
        </th>
        <th>
          status
        </th>
        <th>
          minQty
        </th>
      </tr>
      <tr
        onClick={[Function]}
      >
        <td>
          ETHSGD
        </td>
        <td>
          SGD
        </td>
        <td>
          357
        </td>
        <td>
          
        </td>
        <td>
          ETH
        </td>
        <td>
          ETH
        </td>
        <td>
          0.0000001
        </td>
        <td>
          140
        </td>
        <td>
          0
        </td>
        <td>
          140.0000000
        </td>
        <td>
          -1
        </td>
        <td>
          140.0000000
        </td>
        <td>
          STANDARD
        </td>
        <td>
          140.0000000
        </td>
        <td>
          SGD
        </td>
        <td />
        <td />
        <td>
          0.01
        </td>
        <td>
          2.55
        </td>
        <td>
          10
        </td>
        <td>
          2.5500000
        </td>
        <td>
          8
        </td>
        <td>
          
        </td>
        <td>
          140.0000000
        </td>
        <td>
          TRADING
        </td>
        <td>
          1e-8
        </td>
      </tr>
      <tr
        onClick={[Function]}
      >
        <td>
          BTCSGD
        </td>
        <td>
          SGD
        </td>
        <td>
          180.6
        </td>
        <td>
          ฿
        </td>
        <td>
          Bitcoin
        </td>
        <td>
          BTC
        </td>
        <td>
          0.01
        </td>
        <td>
          53.7
        </td>
        <td>
          0
        </td>
        <td>
          53.70
        </td>
        <td>
          -1
        </td>
        <td>
          1.10
        </td>
        <td>
          STANDARD
        </td>
        <td>
          5.00
        </td>
        <td>
          SGD
        </td>
        <td />
        <td />
        <td>
          0.000001
        </td>
        <td>
          57
        </td>
        <td>
          10
        </td>
        <td>
          57.00
        </td>
        <td>
          8
        </td>
        <td>
          
        </td>
        <td>
          53.70
        </td>
        <td>
          TRADING
        </td>
        <td>
          1e-8
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
